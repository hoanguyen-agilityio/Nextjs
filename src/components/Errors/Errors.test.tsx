import { render, screen } from '@testing-library/react';
import Error from '.';

describe('Error component', () => {
  test('renders error', () => {
    const { container } = render(<Error />);
    expect(container).toMatchSnapshot();
  });

  test('renders with default message and no reset button', () => {
    render(<Error />);
    expect(screen.getByText('Go to Home Page')).toBeInTheDocument();
    expect(
      screen.queryByRole('button', { name: /reset/i }),
    ).not.toBeInTheDocument();
  });

  test('renders with custom message', () => {
    const customMessage = 'A custom error occurred.';
    render(<Error message={customMessage} />);
    expect(screen.getByText(customMessage)).toBeInTheDocument();
    expect(screen.getByText('Go to Home Page')).toBeInTheDocument();
  });

  test('renders with default message and no reset button', () => {
    render(<Error />);
    expect(
      screen.getByText(
        'We apologize for the inconvenience. The page you are looking for might have been removed or is temporarily unavailable.',
      ),
    ).toBeInTheDocument();
    expect(screen.getByText('Go to Home Page')).toBeInTheDocument();
    expect(
      screen.queryByRole('button', { name: /reset/i }),
    ).not.toBeInTheDocument();
  });
});
