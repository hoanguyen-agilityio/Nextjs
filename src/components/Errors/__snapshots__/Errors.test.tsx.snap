// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component renders error 1`] = `
<div>
  <section
    class="min-h-screen flex justify-center"
  >
    <div
      class="max-w-2xl p-5 flex flex-col items-center justify-center m-auto"
    >
      <h1
        class="text-3xl font-bold text-red-500 mb-4 text-center"
      >
        Error 500: Internal server error
      </h1>
      <p
        class="text-red-600 text-xl"
      >
        We apologize for the inconvenience. The page you are looking for might have been removed or is temporarily unavailable.
      </p>
      <a
        aria-label="Go to Home Page"
        class="pt-5 text-ratio-400 text-xl hover:underline"
        href="/"
      >
        Go to Home Page
      </a>
    </div>
  </section>
</div>
`;
